packages/server/lib/plugins/child/run_plugins.js

```diff
@@ -125,6 +125,9 @@ class RunPlugins {
     })
     .then((modifiedCfg) => {
       debug('plugins file successfully loaded')
+      // provide the config globally so plugins can access it
+      global.__CYPRESS_CONFIG__ = modifiedCfg
```

Comment:
Oof....this is quite the obj to put on the process specifically for this change. Once we go GA, this won't necessarily be needed anymore either. Any reason you didn't add as `__CYPRESS_EXPERIMENAL_SESSION_AND_ORIGIN__`? and/or a configuration option that could be pulled when both experimental mode & GA mode will need to be supported?

Also, does the modifiedCfg included the resolved configuration as well? `config.resolved`? if so, we should remove that

Reply:
Could align with naming you used in the preprocessor: `__supportCypressCrossOriginCallback`

Reply:
That's a good point and it helped me recognize a few issues with this approach. 

I refactored things in [`90648a0` (#23283)](https://github.com/cypress-io/cypress/pull/23283/commits/90648a01f517b743f79e9bd3cc2afbc7859b3283) so it no longer sets the config as a global. Instead, the Cypress binary sets a list of commands based on the flag. This accomplishes the following:

- It greatly lightens the size of the value.
- Once we remove the experimental flag, it can just always set the value. Otherwise, we would have still had to set the experimental flag on this global object so it would be backwards-compatible.
- Once we add iframe support, we can expand the list of commands, adding whatever name we decide on for it.